tests/System.Text.Primitives.Tests/ParserTests.tt

```diff
@@ -174,6 +199,68 @@ namespace System.Text.Primitives.Tests
 #>
 		#region <#= signedPrimitives[i] #>
 
+		[Theory]
+        [InlineData("111", true, 0, 111, 3)]
+        [InlineData("blahblahh<#= midValue #>", true, 9, <#= midValue #>, <#= midValue.Length #>)]
+        [InlineData("<#= randValue #>abcdefg", true, 0, <#= randValue #>, <#= randValue.Length #>)]
+        [InlineData("The smallest of this type is <#= minValue #>.", true, 29, <#= minValue #>, <#= minValueLength #>)]
+        [InlineData("Letthem<#= midMinValue #>eatcake", true, 7, <#= midMinValue #>, <#= midMinValue.Length #>)]
+        [InlineData("<#= maxValue #>", true, 0, <#= maxValue #>, <#= maxValueLength #>)] // max
+        [InlineData("<#= minValue #>", true, 0, <#= minValue #>, <#= minValueLength #>)] // min
+        [InlineData("-A", false, 0, 0, 0)] // invalid character after a sign
+        [InlineData("I am 1", false, 0, 0, 0)] // invalid character test
+        [InlineData(" !", false, 0, 0, 0)] // invalid character test w/ char < '0'
+        [InlineData("<#= overflowValue #>", false, 0, 0, 0)] // positive overflow test
+        [InlineData("<#= underflowValue #>", false, 0, 0, 0)] // negative overflow test
+        public unsafe void ParseCustomCultureByteArrayTo<#= capitalizedPrimitive #>(string text, bool expectSuccess, int index, <#= signedPrimitives[i] #> expectedValue, int expectedBytesConsumed)
+        {
+            <#= signedPrimitives[i] #> parsedValue;
+            int bytesConsumed;
+            FormattingData fd = new FormattingData(new byte[] { 48 }, new byte[] { 49 }, new byte[] { 50 }, new byte[] { 51 }, new byte[] { 52 },
+                new byte[] { 53 }, new byte[] { 54 }, new byte[] { 55 }, new byte[] { 56 }, new byte[] { 57 }, new byte[] { 46 }, null,
+                new byte[] { 73, 110, 102, 105, 110, 105, 116, 121 }, new byte[] { 45 }, new byte[] { 43 }, new byte[] { 78, 97, 78 }, new byte[] { 69 }, new byte[] { 101 },
+                FormattingData.Encoding.Utf8, (byte)FormattingData.ParsingStates.PARSING);
+            Format.Parsed nf = new Format.Parsed('G');
+            bool result = InvariantParser.TryParse(UtfEncode(text, false), index, fd, nf, out parsedValue, out bytesConsumed);
+
+            Assert.Equal(expectSuccess, result);
+            Assert.Equal(expectedValue, parsedValue);
+            Assert.Equal(expectedBytesConsumed, bytesConsumed);
+        }
+
+		[Theory]
+        [InlineData("๑๑๑", true, 0, 111, 9)]
```

Comment:
In a future PR, these values will be generated most likely so that they can vary from type to type. Currently, they're all hardcoded just for the sake of being able to test something.